### PR TITLE
Helm Chart improvements

### PR DESCRIPTION
**Pull request by schlapzz**
* Add option to create ServiceAccount
* Add option to reference existing service account
* Add Option to reference existing secrets (often used for GitOps style deployments. eg. External or Sealed Secret)

### CODE REVIEW

**Review by TomChv — APPROVED** [review]
Changes looks fine to me, I let @gerhard double check before merging that one.

**Review by gerhard — COMMENTED** [review]
I could do with some help on this @TomChv 

Could you please try these changes out on a local `kind` cluster. FWIW https://daggerverse.dev/mod/github.com/aweris/daggerverse/kind@a77e1a3398109f46a6d7d01cee33e59dd653a774

File: helm/dagger/Chart.yaml [hunk 1]
```diff
@@ -3,5 +3,5 @@ name: dagger-helm
 description: Dagger Helm chart
 
 type: application
-version: 0.0.2
+version: 0.0.3
 appVersion: v0.8.7
```
[the remark this answers is not in the document]
Comment (schlapzz) [reply helm/dagger/Chart.yaml]: Done

**Comment by TomChv** [comment]
I tried to apply the Helm Chart to the Helm Cluster and it works:

```
➜  dagger git:(main) ✗ helm package .          
Successfully packaged chart and saved it to: /Users/tomchauveau/Documents/DAGGER/dagger/helm/dagger/dagger-helm-0.0.2.tgz
➜  dagger git:(main) ✗ helm install dagger ./dagger-helm-0.0.2.tgz 
NAME: dagger
LAST DEPLOYED: Tue Nov 14 14:29:37 2023
NAMESPACE: default
STATUS: deployed
REVISION: 1
TEST SUITE: None
➜  dagger git:(main) ✗ kubectl get pods
NAME                              READY   STATUS              RESTARTS   AGE
dagger-dagger-helm-engine-fksr4   0/1     ContainerCreating   0          24s
➜  dagger git:(main) ✗ kubectl get pods
NAME                              READY   STATUS    RESTARTS   AGE
dagger-dagger-helm-engine-fksr4   1/1     Running   0          26m
```

Unfortunately, we cannot use the kind for integration test (for now), there are some issue with the object state because he's not return when it should.

I think we can have a follow up of this PR with an actual integration test when the kind module is ready for code usage and not simply dagger call.

> As soon as this looks good, happy to help tag a new helm chart version so that the new version becomes available.

How can I tag the new Helm version?

**Comment by gerhard** [comment]
I was expecting `./dagger-helm-0.0.3.tgz` to be created from the changes in this PR @TomChv.

I intend to make time tomorrow to try this out on a `1.28.2` DKS:
image

**Comment by gerhard** [comment]
I just finished testing this on DO K8s 1.28 and all cases work as expected. I left a bunch of context in my commit messages.

Do you agree with my changes @schlapzz? If yes, I am keen to get this merged and the new version published.

Thank you for sticking with us @schlapzz!

---
image

**Comment by gerhard** [comment]
Re Node.js SDK failures: https://github.com/dagger/dagger/pull/6118#issuecomment-1824432538

**Comment by gerhard** [comment]
Today is a great day to get this merged @schlapzz. WDYT?

**Comment by schlapzz** [comment]
@gerhard Indeed, go for it!

**Comment by gerhard** [comment]
Now available as `registry.dagger.io/dagger-helm:0.1.0`

```sh
helm upgrade --create-namespace --install --namespace dagger dagger oci://registry.dagger.io/dagger-helm
```

**Pull request merged**